### ISSUE TITLE
[Blackhole Bringup] Test Unpack_tilize_A_B

### ISSUE THREAD

**Issue by rtawfik01** [tenstorrent/tt-metal#10498]
This feature: `llk_unpack_tilizeA_B` has been copied over to Blackhole, but is not tested yet, there are no C++ metal tests available for testing it.

It is used in these compute kernels:
```
ttnn/cpp/ttnn/experimental/tt_dnn/op_library/pool/kernels/compute/max_pool_multi_core.cpp
./ttnn/cpp/ttnn/experimental/tt_dnn/op_library/pool/kernels/compute/max_pool.cpp
```

And eager tests:
```
pytest tests/ttnn/unit_tests/operations/test_max_pool2d.py
```

should be calling the above compute kernels. In general, a slow dispatch C++ test should also be made for this operation. This api needs to be verified for max pool functionality.

@mywoodstock has a created a single core version of the above test to help with the testing process:
```
git checkout asarje/maxpool-single-core-test
pytest tests/ttnn/unit_tests/operations/test_max_pool2d.py::test_run_max_pool_single_core[dtype=DataType.BFLOAT16-nblocks=1-dilation=(1, 1)-stride=(2, 2)-padding=(1, 1)-kernel_size=(3, 3)-act_shape=[1, 32, 64, 64]-device_params={'l1_small_size': 24576}]
```

@ttmtrajkovic fyi

**Comment by rtawfik01** [comment]
@ttmtrajkovic This op is needed for maxpool operations since the tilize operation is done on SrcA, while SrcB contains scalars, and then the reduce operation is performed. Currently, with the way tilize operation is implemented for blackhole, tilize is done on both unpack and pack in order to match wormhole b0 tilizing performance. Therefore, we currently cannot do an unpack tilize operation and do compute on the values in dest, since the tilizing is not complete until it reaches packer.

2 possible solutions:
1. Add a tilize llk op for blackhole that only uses the unpacker, but this would likely half of the wormhole performance.
2. Impose restrictions to not fuse the tilize op with other ops for blackhole.

As of right now, I can measure performance of an unfused tilize (with current implementation) + reduce op, and then measure performance of tilize (with only unpack implementation) fused with reduce, and see which option to go forward with.

**Comment by rtawfik01** [comment]
Have Implementation in PR #11572 

Added a unit test for tilize_A_B performance when fused with binary add (for simpler debug and verification). The following is the device performance for the API in the unit test.

Current on device performance values for 16 tiles:
Blackhole: ~17.5k cycles
Wormhole B0: ~17k cycles

Current on device performance values for 64 tiles:
Blackhole: ~51k cycles
Wormhole B0: ~51k cycles

These cycles are the total kernel execution time, and they are around equal for both WHB0 and BH with the cb blocking functions removed as well.

To reproduce:
```
ENABLE_TRACY=1 scripts/build_scripts/build_with_profiler_opt.sh
ninja tests -C build
ENABLE_TRACY=1 TT_METAL_DEVICE_PROFILER=1 TT_METAL_SLOW_DISPATCH_MODE=1 ./build/test/tt_metal/unit_tests --gtest_filter=*ComputeUnpackTilizeA_B*
./tt_metal/tools/profiler/process_device_log.py
```

From the high-level op view on device, seems that tilize_A_B implementation does not affect total op performance, although more in depth investigations are needed with waveforms.

@ttmtrajkovic fyi

**Comment by rtawfik01** [comment]
Added the maxpool fixes here: #11761.
The above maxpool single core tests pass, and most of the maxpool sweep passes, the failures will be documented in a separate issue since some are BH grid size compilation failures: https://github.com/tenstorrent/tt-metal/issues/11684
